pkg/validation/internal/removed_apis.go

```diff
@@ -118,33 +115,84 @@ func validateDeprecatedAPIS(bundle *manifests.Bundle, versionProvided string) (e
 		}
 	}
 
-	// if the k8s value was informed and it is >=1.16 we should check
-	// if the k8s value was not informed we also should check since the
-	// check should occurs with any minKubeVersion value:
-	// - if minKubeVersion empty then means that the project can be installed in any version
-	// - if minKubeVersion any version defined it means that we are considering install
-	// in any upper version from that where the check is always applied
-	if !isVersionProvided || semVerVersionProvided.GE(semVerk8sVerV1betav1Deprecated) {
-		deprecatedAPIs := getRemovedAPIsOn1_22From(bundle)
-		if len(deprecatedAPIs) > 0 {
-			deprecatedAPIsMessage := generateMessageWithDeprecatedAPIs(deprecatedAPIs)
-			// isUnsupported is true only if the key/value OR minKubeVersion were informed and are >= 1.22
-			isUnsupported := semVerVersionProvided.GE(semVerK8sVerV1betav1Unsupported) ||
-				semverMinKube.GE(semVerK8sVerV1betav1Unsupported)
+	// Check if the bundle has removed APIs in 1.22
+	errs, warns = checkRemovedAPIS_1_22(bundle, semVerVersionProvided, semverMinKube, errs, warns)
+
+	// Check if the bundle has removed APIs in 1.25
+	errs, warns = checkRemovedAPIS_1_25(bundle, semVerVersionProvided, semverMinKube, errs, warns)
+
+	// Check if the bundle has removed APIs in 1.26
+	errs, warns = checkRemovedAPIS_1_26(bundle, semVerVersionProvided, semverMinKube, errs, warns)
+
+	return errs, warns
+}
+
+func checkRemovedAPIS_1_22(bundle *manifests.Bundle, semVerVersionProvided semver.Version, semverMinKube semver.Version, errs []error, warns []error) ([]error, []error) {
+	removedAPIs_1_22 := getRemovedAPIsOn1_22From(bundle)
+	if len(removedAPIs_1_22) > 0 {
+		deprecatedAPIsMessage := generateMessageWithDeprecatedAPIs(removedAPIs_1_22)
+		msg := fmt.Errorf("this bundle is using APIs which were deprecated and removed in v1.22. More info: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-22. Migrate the API(s) for %s", deprecatedAPIsMessage)
+		const k8sVersion_1_22 = "1.22.0"
+		semVerK8sVer_1_22 := semver.MustParse(k8sVersion_1_22)
+		if isK8sVersionInformedEQ(semVerVersionProvided, semVerK8sVer_1_22, semverMinKube) {
 			// We only raise an error when the version >= 1.22 was informed via
 			// the k8s key/value option or is specifically defined in the CSV
-			msg := fmt.Errorf("this bundle is using APIs which were deprecated and removed in v1.22. More info: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-22. Migrate the API(s) for %s", deprecatedAPIsMessage)
-			if isUnsupported {
-				errs = append(errs, msg)
-			} else {
-				warns = append(warns, msg)
-			}
+			errs = append(errs, msg)
+		} else {
+			warns = append(warns, msg)
 		}
 	}
+	return errs, warns
+}
 
+func checkRemovedAPIS_1_25(bundle *manifests.Bundle, semVerVersionProvided semver.Version, semverMinKube semver.Version, errs []error, warns []error) ([]error, []error) {
```

Comment:
Feels like this could be one function, checkRemovedAPIsForVersion(..., k8sVersion)